lib/active_model/serializer.rb

```diff
@@ -122,8 +122,12 @@ def initialize(object, options = {})
       end
     end
 
+    def object_model_name
+      object.class.model_name
+    end
+
```

Comment:
using the 'serializer.json_key' is what I suggested in the description of the pull request as an alternative (I had this code already done before I realised this). Using 'serializer.json_key' would probably make more sense, even though the json api adapter then needs to pluralise it.